CDCgov/prime-reportstream
# 11088: upgrade to gradle 8.3

This PR updates gradle to 8.3

Test Steps:
1. Check this branch and refresh the gradle project
2. Run a bunch of different gradle tasks (i.e. quickRun, test, compile, run, fatJar)
3. Start the server and run the smoke tests


## Changes
- Update to gradle 8.3
- Clean up deprecation warnings in the build file

## Checklist

### Testing

### Process

## Linked Issues
- #11088 

## To Be Done
- #9819 
- #9818 

## Specific Security-related subjects a reviewer should pay specific attention to
